File: src/main/java/ch/daplab/nosql/cassandra/doodle/services/impl/dummy/DummyPollServiceImpl.java
package ch.daplab.nosql.cassandra.doodle.services.impl.dummy;

import java.util.*;
import java.util.concurrent.ConcurrentHashMap;

import ch.daplab.nosql.cassandra.doodle.domains.Poll;
import ch.daplab.nosql.cassandra.doodle.domains.Subscriber;
import ch.daplab.nosql.cassandra.doodle.services.PollService;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public class DummyPollServiceImpl implements PollService {

	private static final Logger logger_c = LoggerFactory.getLogger(DummyPollServiceImpl.class);

	private final Map<String, Poll> polls = new ConcurrentHashMap<>();


	@Override
	public Poll getPollById(String pollId) {
		return polls.get(pollId);
	}

	@Override
	public List<Poll> getAllPolls() {
		return new ArrayList<>(polls.values());
	}

	@Override
	public Poll createPoll(Poll poll) {
        String uuid = UUID.randomUUID().toString();
		polls.put(uuid, poll);
        poll.setId(uuid);
		return poll;
	}

	@Override
	public Poll addSubscriber(String pollId, Subscriber subscriber) {
		Poll poll = polls.get(pollId);
		poll.getSubscribers().add(subscriber);
		return poll;
	}

	@Override
	public void deletePoll(String pollId) {
		polls.remove(pollId);
	}

}

File: src/main/kotlin/ch/daplab/nosql/cassandra/doodle/services/impl/dummy/DummyPollServiceImpl.kt
package ch.daplab.nosql.cassandra.doodle.services.impl.dummy

import java.util.*
import java.util.concurrent.ConcurrentHashMap

import ch.daplab.nosql.cassandra.doodle.domains.impl.DataPoll
import ch.daplab.nosql.cassandra.doodle.domains.impl.DataSubscriber
import ch.daplab.nosql.cassandra.doodle.services.PollService
import org.slf4j.Logger
import org.slf4j.LoggerFactory

class DummyPollServiceImpl : PollService {

    private val polls = ConcurrentHashMap<String, DataPoll>()


    override fun getPollById(pollId: String): DataPoll? {
        return polls[pollId]
    }

    override val allDataPolls: List<DataPoll>
        get() = ArrayList(polls.values)

    override fun createPoll(dataPoll: DataPoll): DataPoll {
        val uuid = UUID.randomUUID().toString()
        polls.put(uuid, dataPoll)
        dataPoll.id = uuid
        if (dataPoll.subscribers == null) {
            dataPoll.subscribers = LinkedList<DataSubscriber>()
        }
        return dataPoll
    }

    override fun addSubscriber(pollId: String, dataSubscriber: DataSubscriber): DataPoll {
        val poll = polls[pollId]
        poll!!.subscribers!!.add(dataSubscriber)
        return poll
    }

    override fun deletePoll(pollId: String) {
        polls.remove(pollId)
    }

    companion object {

        private val logger_c = LoggerFactory.getLogger(DummyPollServiceImpl::class.java!!)
    }

}
